src/librustc_mir/transform/qualify_consts.rs

```diff
@@ -529,51 +529,12 @@ impl Qualif for IsNotPromotable {
         let fn_ty = callee.ty(cx.body, cx.tcx);
         match fn_ty.kind {
             ty::FnDef(def_id, _) => {
-                match cx.tcx.fn_sig(def_id).abi() {
-                    Abi::RustIntrinsic |
-                    Abi::PlatformIntrinsic => {
-                        assert!(!cx.tcx.is_const_fn(def_id));
-                        match &cx.tcx.item_name(def_id).as_str()[..] {
-                            | "size_of"
-                            | "min_align_of"
-                            | "needs_drop"
-                            | "type_id"
-                            | "bswap"
-                            | "bitreverse"
-                            | "ctpop"
-                            | "cttz"
-                            | "cttz_nonzero"
-                            | "ctlz"
-                            | "ctlz_nonzero"
-                            | "wrapping_add"
-                            | "wrapping_sub"
-                            | "wrapping_mul"
-                            | "unchecked_shl"
-                            | "unchecked_shr"
-                            | "rotate_left"
-                            | "rotate_right"
-                            | "add_with_overflow"
-                            | "sub_with_overflow"
-                            | "mul_with_overflow"
-                            | "saturating_add"
-                            | "saturating_sub"
-                            | "transmute"
-                            | "simd_insert"
-                            | "simd_extract"
-                            => return true,
-
-                            _ => {}
-                        }
-                    }
-                    _ => {
-                        let is_const_fn =
-                            cx.tcx.is_const_fn(def_id) ||
-                            cx.tcx.is_unstable_const_fn(def_id).is_some() ||
-                            cx.is_const_panic_fn(def_id);
-                        if !is_const_fn {
-                            return true;
-                        }
-                    }
+                let is_const_fn =
+                    cx.tcx.is_const_fn(def_id) ||
+                    cx.tcx.is_unstable_const_fn(def_id).is_some() ||
+                    cx.is_const_panic_fn(def_id);
```

Comment:
Isn't this `IsNotPromotable`? Shouldn't that check the `rustc_promotable` attribute instead of allowing *all* `const fn`?

Reply:
Ah dang I might have mixed this up again with `IsNotImplicitlyPromotable`...

Reply:
Note that this can't go wrong anymore, without triggering a mismatch with `promote_consts`' copy of the logic (although we should make sure this PR updates both properly), which would ICE.

@ecstatic-morse and I will be removing the copy from `qualify_consts` in a week or two (maybe we just need to wait for beta to branch?), so `IsNot(Implicitly) Promotable` is going away soon!

Reply:
I just hope we get the intrinsic thing landed as part of this refactor, it's been 4 months or so.^^

Reply:
Oh yeah this PR can land at any time, and I think `promote_consts` even has the right code already, without another copy of the list of intrinsics.

Reply:
Okay, but what about the new const checking pass? Does that still allow calling intrinsics?

Reply:
That's a good question for @ecstatic-morse.

Reply:
I reserved a spot for these checks [here](https://github.com/rust-lang/rust/blob/b43a6822597061dc18cbdde1769d9815e718d7bb/src/librustc_mir/transform/check_consts/validation.rs#L512). I think we just need to call the `is_const_intrinsic` from that spot? I dunno if this PR makes additional changes to the validation logic though.

Reply:
I missed that this was baked into `in_const_fn` now (see below), I think this PR just needs to remove that `return` along with the `FIXME` above it.